edit/edit.js

```diff
@@ -508,7 +517,7 @@ window.onbeforeunload = () => {
   if (isCleanGlobal()) {
     return;
   }
-  updateLintReport(null, 0);
+  updateLintReportIfEnabled(null, 0);
   return confirm(t('styleChangesNotSaved'));
```

Comment:
LOL, ok... I've been kind of stuck on this one anyway... because `messageBox` returns a promise and `onbeforeunload` is expecting a boolean, it gets a lot more complicated.